File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcDistributionElementType/IfcDistributionFlowElementType/IfcEnergyConversionDeviceType/IfcTransformerType/IfcTransformerType.cs
public class IfcTransformerType : IfcEnergyConversionDeviceType
{
    public IfcTransformerTypeEnum PredefinedType { get; set; }

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : (PredefinedType <> IfcTransformerTypeEnum.USERDEFINED) OR ((PredefinedType = IfcTransformerTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
}

File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcDistributionElementType/IfcDistributionFlowElementType/IfcEnergyConversionDeviceType/IfcTransformerType/IfcTransformerType.vb
Public Class IfcTransformerType Inherits IfcEnergyConversionDeviceType
    Public PredefinedType As IfcTransformerTypeEnum

    ' === WHERE CLAUSES ===
    ' CorrectPredefinedType : (PredefinedType <> IfcTransformerTypeEnum.USERDEFINED) OR ((PredefinedType = IfcTransformerTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
End Class
